src/Algebra/Construct/Pointwise.agda

```diff
@@ -0,0 +1,224 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- For each `IsX` algebraic structure `S`, lift the structure to the
+-- 'pointwise' function space `A → S`: categorically, this is the
+-- *power* object in the relevant category of `X` objects and morphisms
+--
+-- NB the module is parametrised only wrt `A`
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Algebra.Construct.Pointwise {a} (A : Set a) where
+
+open import Algebra.Bundles
+open import Algebra.Core using (Op₁; Op₂)
+open import Algebra.Structures
+open import Data.Product.Base using (_,_)
+open import Function.Base using (id; _∘′_; const)
+open import Level
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Bundles using (Setoid)
+open import Relation.Binary.Structures using (IsEquivalence)
+
+
+private
+
+  variable
+    c ℓ : Level
+    C : Set c
+    _≈_ : Rel C ℓ
+    ε 0# 1# : C
+    _⁻¹ -_ : Op₁ C
+    _∙_ _+_ _*_ : Op₂ C
+
+  lift₀ : C → A → C
+  lift₀ = const
+
+  lift₁ : Op₁ C → Op₁ (A → C)
+  lift₁ = _∘′_
+
+  lift₂ : Op₂ C → Op₂ (A → C)
+  lift₂ _∙_ g h x = (g x) ∙ (h x)
+
+  liftRel : Rel C ℓ → Rel (A → C) (a ⊔ ℓ)
+  liftRel _≈_ g h = ∀ x → (g x) ≈ (h x)
+
+
+------------------------------------------------------------------------
+-- Setoid structure: here rather than elsewhere? (could be imported?)
+
+module _ (isEquivalenceC : IsEquivalence _≈_) where
+
+  open IsEquivalence isEquivalenceC
+
+  isEquivalence : IsEquivalence (liftRel _≈_)
+  isEquivalence = record
+    { refl = λ {f} x → refl {f x}
+    ; sym = λ f≈g x → sym (f≈g x)
+    ; trans = λ f≈g g≈h x → trans (f≈g x) (g≈h x)
+    }
+
+------------------------------------------------------------------------
+-- Structures
+
+module _ (isMagmaC : IsMagma _≈_ _∙_) where
```

Comment:
Having anonymous modules for one definition feels overkill to me here.

Reply:
Yes, it worked out fine for the `Bundle`s, but I'd wanted the `private module ...` trick, and didn't easily see how to do that simply in a function block, but perhaps that's where `where` would be my friend. Will refactor!